### PR TITLE
feat(storage): add optional sanity check for committed SSTs

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Now when SSTs are committed to meta, meta can be configured to verify the validity of these SSTs, in terms of Ordered, Locally unique and Globally unique mentioned in src/storage/src/hummock/state_store.rs. 

- The verification is done in a random compactor node.
- This sanity check is disabled by default.

Limitation:
- To keep it simple, when violation spotted, we just panic the compactor node, rather than report it somewhere. Because we will only enable this sanity check in troubleshooting, I think it's enough to record the violation in compactor's log.
- Because the sanity check doesn't prevent committed SSTs from being vacuumed, it's possible some SSTs are deleted before or during sanity check. We just warn and skip these SSTs.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: risedev.yml [hunk 1]
```diff
@@ -538,6 +538,9 @@ template:
     # 7 days by default.
     min-sst-retention-time-sec: 604800
 
+    # Enable sanity check when SSTs are committed. Disabled ny default.
```
Comment (BugenZhao) [inline risedev.yml:541]:
```suggestion
    # Enable sanity check when SSTs are committed. Disabled by default.
```

File: src/storage/src/hummock/validator.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            Arc::new(SstableIteratorReadOptions { prefetch: false }),
+        );
+        let mut previous_key: Option<Vec<u8>> = None;
+        if let Err(err) = iter.rewind().await {
+            tracing::warn!("Skip sanity check for SST {}. {:#?}", sst_id, err);
+        }
+        while iter.is_valid() {
+            key_counts += 1;
+            let current_key = iter.key().to_vec();
+            // Locally unique and Globally unique
```
Comment (BugenZhao) [inline src/storage/src/hummock/validator.rs:67]: Does the `globally` mean "all sstables in this newly commited epoch"?

**Review by BugenZhao — APPROVED** [review]
LGTM! 🚀

File: risedev.yml (hunk 1 above)
Comment (BugenZhao) [reply risedev.yml:541]: How about enable this option in some CI workflows?

File: src/storage/src/hummock/validator.rs (hunk 2 above)
Comment (zwang28) [reply src/storage/src/hummock/validator.rs:67]: Yes

File: risedev.yml (hunk 1 above)
Comment (zwang28) [reply risedev.yml]: Yes, I've enabled it for some risedev config.
The option doesn't affect overall CI execution time.

**Pull request merged**